test/e2e/apimachinery/webhook.go

```diff
@@ -244,6 +246,236 @@ var _ = SIGDescribe("AdmissionWebhook", func() {
 		slowWebhookCleanup()
 	})
 
+	ginkgo.It("patching/updating a validating webhook should work", func() {
+		client := f.ClientSet
+		admissionClient := client.AdmissionregistrationV1()
+
+		ginkgo.By("Creating a validating webhook configuration")
+		hook, err := createValidatingWebhookConfiguration(f, &admissionregistrationv1.ValidatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: f.UniqueName,
+			},
+			Webhooks: []admissionregistrationv1.ValidatingWebhook{
+				newDenyConfigMapWebhookFixture(f, context),
+			},
+		})
+		framework.ExpectNoError(err, "Creating validating webhook configuration")
+		defer func() {
+			err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Delete(hook.Name, nil)
+			framework.ExpectNoError(err, "Deleting validating webhook configuration")
+		}()
+
+		hook, err = admissionClient.ValidatingWebhookConfigurations().Get(f.UniqueName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Getting validating webhook configuration")
+		ginkgo.By("Updating a validating webhook configuration's rules to not include the create operation")
+		hook.Webhooks[0].Rules[0].Operations = []admissionregistrationv1.OperationType{admissionregistrationv1.Update}
+		hook, err = admissionClient.ValidatingWebhookConfigurations().Update(hook)
+		framework.ExpectNoError(err, "Updating validating webhook configuration")
+
+		ginkgo.By("Creating a configMap that does not comply to the validation webhook rules")
+		err = wait.PollImmediate(100*time.Millisecond, 30*time.Second, func() (bool, error) {
+			cm := namedNonCompliantConfigMap(string(uuid.NewUUID()), f)
+			_, err = client.CoreV1().ConfigMaps(f.Namespace.Name).Create(cm)
+			if err != nil {
+				if !strings.Contains(err.Error(), "denied") {
+					return false, err
+				}
+				return false, nil
+			}
+			return true, nil
+		})
+		framework.ExpectNoError(err, "Waiting for configMap in namespace %s to be allowed creation since webhook was updated to not validate create", f.Namespace.Name)
+
+		ginkgo.By("Patching a validating webhook configuration's rules to include the create operation")
+		hook, err = admissionClient.ValidatingWebhookConfigurations().Patch(
+			f.UniqueName,
+			types.JSONPatchType,
+			[]byte(`[{"op": "replace", "path": "spec.webhooks[0].rules[0].operations", "value": ["CREATE"]}]`))
```

Comment:
shouldn't path be `/spec/webhooks/0/rules/0/operations`?

Reply:
Oddly, only `spec/webhooks/0/rules/0/operations` (without the proceeding `/`) works. This seems wrong and I don't know why, but it works. With the `/` prefix, I get an error like `Invalid ""`.

Reply:
there's no spec field... this should actually be `/webhooks/0/rules/0/operations`

it looks like the jsonpatch apply code does not check there is a leading slash and silently discards anything that precedes a slash here:

https://github.com/kubernetes/kubernetes/blob/f6a70ef27163d1171412b6f97049968896b8d1f9/vendor/github.com/evanphx/json-patch/patch.go#L307-L313

That's a bug in the jsonpatch library which is too liberal with the paths it accepts. Opened https://github.com/kubernetes/kubernetes/issues/81422 and https://github.com/evanphx/json-patch/issues/86 to track that. For this PR, fix the paths to `/webhooks/...`

Reply:
Woah, in my fumbling, looks like we found something. Thanks for filing! Fixing this test to use `/webhooks`